### DISCUSSION TITLE
How does homebrew learn about software version?

### DISCUSSION THREAD

**Discussion by o2genum** [Homebrew/discussions#1115] [Getting started] [answered]
I see, Homebrew is aware of software versions:
```
$ tree -L 2 /opt/homebrew/Cellar 
/opt/homebrew/Cellar
├── autoconf
│   └── 2.69
├── bdw-gc
│   └── 8.0.4_2
├── c-ares
│   └── 1.17.1
├── gdbm
│   ├── 1.18.1_1
│   └── 1.19
├── gettext
│   └── 0.21
...
```

But where does it get these versions?

The Formulae seem to include:
- big versions allowing for parallel installs like `python@3`
- `revision` field

... but not the kind of version I see in the Cellar directory tree.

**Comment by SMillerDev** [comment, marked as answer]
It parses it from the URL or uses a `version` field if that fails.
